rioxarray/_io.py

```diff
@@ -711,6 +796,49 @@ def _load_subdatasets(
     return dataset
 
 
+def _load_bands_as_variables(
+    riods: RasterioReader,
+    parse_coordinates: bool,
+    chunks: Optional[Union[int, Tuple, Dict]],
+    cache: Optional[bool],
+    lock: Any,
+    masked: bool,
+    mask_and_scale: bool,
+    decode_times: bool,
+    decode_timedelta: Optional[bool],
+    **open_kwargs,
+) -> Union[Dataset, List[Dataset]]:
+    """
+    Load in rasterio bands as variables
+    """
+    global_tags = _parse_tags(riods.tags())
+    data_vars = {}
+    for band in riods.indexes:
+        band_riods = SingleBandDatasetReader(
+            riods=riods,
+            bidx=band - 1,
+        )
+        band_name = f"band_{band}"
+        data_vars[band_name] = (
```

Comment:
Why not leaving the possibility to use band description as band name if it is available?

Reply:
The variable name should only contain alphanumeric characters and underscores. The band description could potentially be a sentence with any characters. This ensures consistency and stability.

Reply:
Fair. Then in my scenario I will just use the variables/band metadata and set their names accordingly.
Important is that we parse also the bands metadata and now seems fine!